### PR TITLE
Allow Stepmania 5.3

### PR DESCRIPTION
**Pull request by concubidated**
5.3 works perfectly with SL and there doesn't seem to be a valid reason to manually block it.

### CODE REVIEW

**Comment by quietly-turning** [comment]
Thanks for this pull request.

> 5.3 works perfectly with SL

[Past pull requests to this project](https://github.com/quietly-turning/Simply-Love-SM5/pull/114) and recent [work being done elsewhere](https://github.com/48productions/Simply-Potato-SM5/commit/138535be5a4c13d981380d78b0ca4467a47fe72c#commitcomment-38021734) suggest that this is not the case.

> there doesn't seem to be a valid reason to manually block it.

I am blocking SM5.3 while it remains closed source to maintain my own sanity as the sole human responsible for maintaining and advancing this project and the community it supports.

I do hear positive rumblings about SM5.3, and I will work to assess what in Simply Love should change–and what will need to change–once it is opened up to the public.

**Comment by Scraticus** [comment]
Hiya Quietly-Turning.

I have noticed that you have coded another check to force users off of using SL. When we agreed this project we promised to support all active content and themes open to 5.1 and 5.0.

I understand you do not wish to involve or talk to us, I have reached out on several occasions to try to do this. However with this block and 'change theme' we are actually unable to test Simply Love, and it shows that as a fellow SM developer, you are not supporting our work at all. I am open to collaboration, as are the whole of my team, but if you do not wish for 5.3 to work with this branch, should i just fork a version that will have the checks removed so you do not need to 'worry' about it moving forwards?

It's difficult enough with all the hate we get from other people out there to find positive time and focus to work on this game, but to get it from supposed fellow SM Dev team members, it's taking the cake a little.

If you want us to fork our own version of SL while we're building a better game, let me know. 

Squirrel

**Comment by SheepyChris** [comment]
Latest version of the theme (4.9) works without any errors whatsoever after applying this patch for StepMania 5.3.

**Comment by quietly-turning** [comment]
Hi @Scraticus,

Thanks for reaching out.  I do appreciate the opportunity to talk with you about this here.

I met with one of the developers (I only have a Facebook name, not a GitHub handle) for your project in December 2017.  My two takeaways from that meeting were:

1. The goal of your project was to use StepMania's existing code to develop a general-purpose, commercial game engine.  I inferred that the code would remain closed source.

2. The primary impetus behind the project was anger.  I asked what the driving force was, what would sustain it when inevitable developer burnout occurred, and the response was, "fuck Konami."

In purely practical terms, it's not a good use of my free time to develop open source software for closed source software.  Figuring out how things work in StepMania is challenging enough even with full access to the source code.

The second point is more troubling, however, and it immediately suggested to me an unbridgeable disconnect in developer culture and personal visions.

In fairness, December 2017 was some time ago.  At that time, I fully believed I would release Simply Love v4.7 and then be done.  It didn't play out that way; I'm still here developing for Simply Love and the StepMania project.

I changed my understanding of something I fully believed then, and while I have not kept up on your project since then, I'd like to believe it's possible for a similar change/growth to have occurred there.

Can you talk about what has changed with your project since December 2017?   I'm specifically interested in learning more about the two points above.

**Comment by Scraticus** [comment]
> Hi @Scraticus,
> 
> Thanks for reaching out. I do appreciate the opportunity to talk with you about this here.
> 
> I met with one of the developers (I only have a Facebook name, not a GitHub handle) for your project in December 2017. My two takeaways from that meeting were:
> 
> 1. The goal of your project was to use StepMania's existing code to develop a general-purpose, commercial game engine.  I inferred that the code would remain closed source.
> 2. The primary impetus behind the project was anger.  I asked what the driving force was, what would sustain it when inevitable developer burnout occurred, and the response was, "fuck Konami."
> 
> In purely practical terms, it's not a good use of my free time to develop open source software for closed source software. Figuring out how things work in StepMania is challenging enough even with full access to the source code.
> 
> The second point is more troubling, however, and it immediately suggested to me an unbridgeable disconnect in developer culture and personal visions.
> 
> In fairness, December 2017 was some time ago. At that time, I fully believed I would release Simply Love v4.7 and then be done. It didn't play out that way; I'm still here developing for Simply Love and the StepMania project.
> 
> I changed my understanding of something I fully believed then, and while I have not kept up on your project since then, I'd like to believe it's possible for a similar change/growth to have occurred there.
> 
> Can you talk about what has changed with your project since December 2017? I'm specifically interested in learning more about the two points above.

Ok, lets get a few things in perspective.

Team Rizu did not start as a project until April 2018, We had not even begun to plan what we would become until late 2018. We formally created 'Project OutFox' in mid 2019, with a rebase of 5.1, and the work we completed from summer 2018.

Whomever that was that spoke to you did not belong to our project, nor spoke for what we planned or had planned to do.

I have never said or 'inferred' a closed source project, we are simply rebasing a lot of work to allow for a stable release when this engine work has been completed. There has never been a 'commercial' project based off of the 5.3 project, and even if there was, it would not breach the agreement we made with shake on keeping things for the community.

The fact you have taken one users opinion and interaction to base your entire understanding when you know how to reach us is frankly shocking. You have also made a point to 'avoid' working with 5.3 where you can based off of one person who definitely did not speak on my behalf. 

The project is moving along steadily, however like with SSC, we will opensource when enough is done to prevent needless issues and bug reports (and the commit rush is stable). We also have never said the words 'fuck konami', that is not our way. The project holds no political allegiance as we are not doing this for status or a point. I have to call out where things are wrong, like this situation. If i had not done this, we would not be having this conversation. 

I would invite you to in future, to get your information from the source, no-one other than me, jousway or cube, speak for team rizu. Anyone else is using their own opinions to clearly ruin your day and make development decisions which are negatively impacting your theme.

Squirrel

**Comment by quietly-turning** [comment]
Thanks for responding.  

It seems like I've misunderstood the relationship between the StepMania project and Team Rizu based on assumptions I've  made, which were largely based on the December 2017 meeting I wrote about above.

I'm willing to throw out my existing understanding of SM5.3 and Team Rizu as it may not be at all accurate, and start fresh.

How can I learn more about SM5.3?  I'm legitimately unfamiliar with the details of your agreement with shakesoda; I hadn't appreciated there was one.

**Comment by Scraticus** [comment]
You are but one of many to make incorrect assumptions and act on them, so we are not offended.

Our developer website is at https://projectmoon.dance where you can find our build history and a bit about the project, as well as our discord server you are welcome to join. We host an inclusive community focusing on all game modes, so it is a wide and friendly community.

We did not have any misunderstandings with you or your project other than why you went out of your way to block the build when we're in alpha and need to test things, hence why I suggested the fork request of this project. We don't openly hold malice to anyone, there have been many misunderstandings involving the project and it's direction, and I am glad this was something we were able to save and I hope to nurture it into a team relationship moving forward.

The project began as a method to add online and profiles to SM5, but when we looked into the engine to see how this could be appropriated it was discovered just how much in a sorry state the code base was, and it looked mostly abandoned. I spoke to Shakesoda in hopes of a resurrection in the form of a team project, and it was requested rather than breaking 10 years of content and community to take the time to make a proper 5.x build that is stable and supports what the community as a whole wants from the game. This backward compatibility support has been an absolute git to do, but it's something now we are down the line which we are proud of, as it has brought many players and newcomers to the community, and provided a very stable experience for folks during the lockdown. We will continue to maintain a near 100% compatibility where we can, and this is taken into consideration whenever we plan a new feature or change engine components.

We asked if it was ok to close the source while this process was done, and it was deemed wise for the health of the repo and the need of people who are just wanting to play. Its why we've been around the repo but not _on_ it as taint can happen and we do not wish this. SSC did this same behaviour before us, so we did not see it as a problem. Some vocal members of the community disagreed, including some of the SMDev Team. We have also been compared to past SM repo contributors who may have left a less than savoury memory in the minds of some, and it has also been hard to differentiate from history we have no involvement with.

I therefore invite you to the Moondance discord for a chat and to formally set forth a lasting team project, as stepmania has had enough animosity and misunderstandings and assumptions to last a lifetime. OutFox was created to move beyond this and to make it's future bright and relevant in this decade.

Squirrel.

**Comment by quietly-turning** [comment]
Thanks, @Scraticus.  I'll take a look at https://projectmoon.dance.  It's been a while since I last looked at it.

I appreciate the invitations to join your project's Discord server.  Generally speaking, I prefer public GitHub threads like this for project discussions as they can be easily linked to and viewed by anyone for future reference.  I think the public nature of it also helps reduce the likelihood that future misunderstandings might occur.

I also acknowledge that working publicly on free, open source software is stressful.  As you pointed out, there can be

> needless issues and bug reports

that whittle away at energy, productivity, and mental health.  I worked on Simply Love v4.8 completely in private from late April to early July 2018 for these reasons, so I understand the desire to occasionally apply intense focus to work without interruption.

In any event, I realistically need some time to rest post-SLv4.9, review your project's website, and collect my thoughts before I can respond helpfully here.

I'm happy that you reached out and hopeful that good things can come of this in the future.

**Comment by quietly-turning** [comment]
Let's reopen this and keep communication open.

As I wrote above, I really need some time to rest, and not work on Simply Love.  I look forward to familiarizing myself with   https://projectmoon.dance in the upcoming weeks. 👍

**Comment by Scraticus** [comment]
Indeed, I am glad the culture of 'SL doesn't work on outfox so do not use outfox' which you have created can be finally undone.

We have no problems taking over issue tracking for any issues from outfox users while we're in alpha, we have our own bug tracker now @ https://github.com/TeamRizu/OutFox where people can report issues with our game.

I feel before you 'rest' you take a moment to reconsider the forced blocks you added in 4.9 as it is ruining player experience and still punishing our development for your mistaken considerations.

Indeed, I am glad we are able to have a positive resolution, but there is still a wrong that needs to be put right, as we are being penalised for nothing other than 'existing' right now.

I too look forward to your return, but for now, enjoy your rest and consider my request.

Squirrel

**Comment by Scraticus** [comment]
Hiya Quietly-Turning, I would like to apologise for my tone in the above posts, I was having a particularly bad day yesterday, and my frustration with this problem ruled my emotions when I constructed that. 

I can see this was all a misunderstanding, and that there was no malicious intent. I let people who think it's fun to harass and hassle us "because SL doesn't work and we're trying to profit off of Quietly Turnings work and community contributions" get to me more than I should have.

I am happy we have an open channel of communication, and I do not wish to taint that with past actions. We can show those who mistakenly think we are corporate and for profit clearer with active collaboration and support so we can build a brighter future for both of our projects.

Be well and have a good rest.

Squirrel

**Comment by quietly-turning** [comment]
To check in on this since it's been a while: I have not yet made time to review https://projectmoon.dance

I have not been doing much with Simply Love since the 4.9 release, but I promise I'll review the website and host an open discussion here, in this thread, before the next Simply Love release occurs.

Committing to an OSS timeline is fraught with peril and burnout, so all I can say right now is "maybe in 2020."

It is genuinely my hope and intent to get a more accurate understanding of your project so we can discuss our respective wishes and priorities honestly and productively.

**Comment by quietly-turning** [comment]
Hi @Scraticus, thanks for giving me the time I needed to rest, reflect, and review https://projectmoon.dance.  I appreciate it. 🙂 

Based on a review of https://projectmoon.dance, it looks like your work continues, and that's good!  These things take time.  

On my end, it's important to me that StepMania, and my contributions to it, remain free/non-commercial.  My stance on this is informed by modern Dance Dance Revolution having extremely limited global availability.  It seems unfair that a person's ability to enjoy DDR is really a matter of access, which itself comes down to what part of what country they happened to be born in.

In this context, I appreciate that StepMania can be installed and enjoyed by nearly *anyone* with a computer, regardless of their PC specs, regardless of whether they live in southern California or South Africa.  A significant part of this is that StepMania is free.

Although no open source license I'm aware of can really *enforce* non-commercial use, I [recently](https://github.com/quietly-turning/Simply-Love-SM5/commit/b3a278e7f7dffa9ff3bb3c998ebc5e35d06cbb4f) switched Simply-Love-SM5's license from MIT to GPLv3 in an effort to encourage it.  It is important to me that StepMania, and my contributions to it, remain free.

One of the FAQs at https://projectmoon.dance/faq currently states that your project will eventually be released under the [Apache License 2.0](https://choosealicense.com/licenses/apache-2.0/).

> Right now we are not maintaining a public code repository, but we will open one in the future (with our code licensed under the Apache License 2.0). 

Screen Shot 2020-10-27 at 8 45 25 PM

I'd like to learn more about what this means, specifically regarding your team's plans for commercial licensing.

For example, do you plan to maintain a paid, closed source product and a free, open source version simultaneously?  Or, do you plan to have a commercial product with some free, open source API, along the lines of the [StepManiaX dance platform SDK](https://steprevolution.github.io/stepmaniax-sdk/)?  Or, do you plan to have a commercial product that is free to use with limitations, or one that features in-app purchases?

In short: do you plan to charge money for any aspect of your project?

**Comment by Scraticus** [comment]
Once again, I am not sure where you have got this commercial licencing idea from. Nowhere on our site does it say we are going to go commercial, nor does it say we will be charging for the game. So lets get that straight first.

**We are not, and have no plans to charge for 5.3** Secondly, you are overstepping your bounds as an open source developer in several of those questions. But for the sake of closing this once and for all, I will answer them.

The reason we are releasing 5.3 (WHICH WILL BE ON THE STEPMANIA REPO) under Apache 2.0 is so the licencing and copyrights and credits to the past contributors remains intact for anyone who takes a fork of 5.3. Project OutFox is our working 'project' like you have for 'Simply Love'.

We have a Patreon which supports our work via Donations, which we do not put any features behind a paywall. Even if we want to release a commercial product in the future, that is beyond your request as an open source themer for one specific version of the game.

**In short: OutFox/5.3 will be merged back into the StepMania Repo and will replace 5.1 as the main branch, and this will be done for free like the way Etterna are doing things.**

If we decide to offer paid for DLC packs of songs/content, they will not affect any availability of the game to anyone. Our alphas are free and available to anyone as you can clearly read. Questioning us on any future commercial projects is beyond the scope of this discussion, However, we do not plan to make a commercial game out of a version of StepMania that would accept your work/other themes anyway. Your work would continue to be free and not monetised by our group, as we have no plans for that. If we do decide to go the way of being a commercial game company, it would likely be on a completely new game, not this project, as this was always about giving back to the community, and to bring the SM players together once more.

Now can you get this damned idea we are 'commercial/for profit' out of your head.

I won't be able to reply to your comment for a around a week to 10 days as I have surgery, please do not think me rude if it takes a few days to respond.

Take care,

Squirrel

**Comment by quietly-turning** [comment]
> **We are not, and have no plans to charge for 5.3**

> we are releasing 5.3 (WHICH WILL BE ON THE STEPMANIA REPO) under Apache 2.0

Thank you for responding.  I really appreciate that you've taken the time to explicitly clarify that SM5.3 will be both:

 * free (good for the global player community)
 * open source (good for developers like me)

Both of these points are important to me in the context of StepMania and StepMania-development.

---

With regard to this pull request, and lifting Simply-Love-SM5's block on SM5.3, my plan for the next Simply-Love-SM5 release is to:

1. dedicate a section of release notes to why I think being free and open source are StepMania's strengths
2. explicitly address why I've blocked SM5.3 to-date, citing [initial misunderstanding](https://github.com/quietly-turning/Simply-Love-SM5/pull/184#issuecomment-657940797), and pointing readers to this thread (https://github.com/quietly-turning/Simply-Love-SM5/pull/184) so they can learn more if they are interested
3. provide instructions on how players can edit `./Scripts/SL-SupportHelpers.lua` to use Simply-Love-SM5 with SM5.3 right now if they wish
4. explicitly clarify that I'll remove the block and "work to assess what in Simply Love should change" when SM5.3 is made open source

**Comment by natano** [comment]
The status of SM 5.3 support in SL is now tracked in https://github.com/Simply-Love/Simply-Love-SM5/issues/214, so closing this PR.

**Comment by RollingStar** [comment]
>  Secondly, you are overstepping your bounds as an open source developer in several of those questions.

This is an incredibly unprofessional sentence from @Scraticus. You are the one overstepping your bounds as an (allegedly) open source developer. It's perfectly reasonable to ask if one's free labor will be appropriated into a commercial project. There's a long history here: XBMC/Kodi into Plex, TiVo using DRM to prevent modified GPL software from running on its hardware, arguably Linux into Android into Google-Android. As you are aware, there are a wide array of open licenses to choose from which have different levels of commercial feasibility. Stepmania's permissive MIT license would leave you in a great position to develop commercial variations of Stepmania / Stepmania 5.3 / OutFox. Even the Apache 2.0 license would leave [commercial work](https://github.com/tensorflow/tensorflow/blob/master/LICENSE) quite possible.

The Android comparison doesn't inspire confidence, in fact, it raises suspicion. I'd feel more comfortable if you just came out and said you were thinking of commercializing the work, instead of talking indirectly about what you're trying to do. You come across like a corporate entity rather than a passionate developer, even though I'm sure you are the latter.

> that is beyond your request as an open source themer for one specific version of the game.

That's weird, because the fact that you're commenting at all implies that Simplylove / Quietly-Turning has something you want. If they offered nothing of value, you wouldn't need to talk to them at all.

> If we decide to offer paid for DLC packs of songs/content, they will not affect any availability of the game to anyone.

Songs and content are very different. Again, you could just say "we're thinking of making paid step packs, but the base game and the source code will be free and open". "Content" could be anything from songs, to themes, to bug fixes, to online play.

> Our alphas are free and available to anyone as you can clearly read.

Stuff like this just reeks of an attempt to bring an open-source project into a closed-source commercialization. It's free until it's not. The source is what matters, not the binaries for Stepmania fork #121251. I'll believe it when I see the source code you promise you'll release.

---

As a developer, if I wanted to work on Stepmania, the software you allegedly care about, I would want to work on a version newer than 2 years old and not waste my time. I have no way of knowing if you've already implemented [this feature](https://github.com/stepmania/stepmania/discussions/2112) I want to implement. And if I add it to baseline Stepmania, I have no way of knowing if it will seamlessly integrate into the code you promise you will release.

This real concern, against your concern about people filing bug reports for alpha software, is not remotely on the same level. You could simply ban people from your discussions for filing bug reports about home-compiled versions, the way you do now for "begging for a source release". You could do all the development in private and not care if Generic Player #1000 has an issue with the game. Anyone able to compile releases on their own probably has some awareness of the software development process, too.

I eagerly await your source code release to prove me wrong. It's a shame I can't develop for Stepmania without potentially wasting my time. I would like to solve that burn-in issue sooner rather than later...

**Comment by concubidated** [comment]
@RollingStar this was completely unnecessary and not helpful in anyway. This issue was closed months ago. No need to vent here.
